modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/display/context/HeaderFragmentDisplayContext.java

```diff
@@ -0,0 +1,509 @@
+				id = "requestQuote";
+				label = "request-a-quote";
+			}
+			else if (commerceOrderStatus.getKey() ==
+						CommerceOrderConstants.ORDER_STATUS_PROCESSING) {
+
+				if (!hasPermission(
+						CommerceOrderActionKeys.APPROVE_OPEN_COMMERCE_ORDERS)) {
+
+					continue;
+				}
+
+				label = "accept-order";
+			}
+			else if (commerceOrderStatus.getKey() ==
+						CommerceOrderConstants.ORDER_STATUS_QUOTE_PROCESSED) {
+
+				label = "process-quote";
+			}
+			else {
+				continue;
+			}
+
+			if (Validator.isNull(buttonCssClass)) {
+				buttonCssClass = "btn-primary";
+			}
+
+			if (commerceOrderStatus.getPriority() ==
+					CommerceOrderConstants.ORDER_STATUS_ANY) {
+
+				buttonCssClass = "btn-secondary";
+			}
+
+			headerActionModels.add(
+				new HeaderActionModel(buttonCssClass, null, href, id, label));
+		}
+
+		return headerActionModels;
+	}
+
+	public String getOrderDetailURL() throws PortalException {
+		PortletURL portletURL = null;
+
+		long groupId = _cpRequestHelper.getScopeGroupId();
+
+		// Identify correct order details url and just use that as
+		// the href rather than doing below plid
+		// search for OPEN_ORDER_CONTENT portlet
+
+		long plid = _portal.getPlidFromPortletId(
+			groupId, CommercePortletKeys.COMMERCE_OPEN_ORDER_CONTENT);
+
+		if (plid > 0) {
+			portletURL = PortletProviderUtil.getPortletURL(
+				_httpServletRequest, CommerceOrder.class.getName(),
+				PortletProvider.Action.EDIT);
+
+			if (getCommerceOrder() != null) {
+				portletURL.setParameter(
+					"mvcRenderCommandName",
+					"/commerce_open_order_content/edit_commerce_order");
+				portletURL.setParameter(
+					"commerceOrderId", String.valueOf(getCommerceOrderId()));
+
+				String backURL = ParamUtil.getString(
+					_httpServletRequest, "backURL");
+
+				portletURL.setParameter("backURL", backURL);
+			}
+
+			return portletURL.toString();
+		}
+
+		plid = _portal.getPlidFromPortletId(
+			groupId, CommercePortletKeys.COMMERCE_CART_CONTENT);
+
+		if (plid > 0) {
+			portletURL = PortletURLBuilder.createLiferayPortletURL(
+				_cpRequestHelper.getLiferayPortletResponse(), plid,
+				CommercePortletKeys.COMMERCE_CART_CONTENT,
+				PortletRequest.RENDER_PHASE
+			).setParameter(
+				"commerceOrderId", getCommerceOrderId()
+			).buildPortletURL();
+		}
+
+		return portletURL.toString();
+	}
+
+	public String getRetryPaymentURL() throws PortalException {
+		return PortletURLBuilder.create(
+			_commerceOrderHttpHelper.getCommerceCheckoutPortletURL(
+				_cpRequestHelper.getRequest())
+		).setParameter(
+			"checkoutStepName", "payment-process"
+		).setParameter(
+			"commerceOrderUuid",
+			() -> {
+				CommerceOrder commerceOrder = getCommerceOrder();
+
+				return commerceOrder.getUuid();
+			}
+		).buildString();
+	}
+
+	public PortletURL getTransitionOrderPortletURL(
+		CommerceOrder commerceOrder) {
+
+		return PortletURLBuilder.createActionURL(
+			_cpRequestHelper.getLiferayPortletResponse()
+		).setActionName(
```

Comment:
`portletResponse` is always null. This needs to be investigated to see if it is actually impossible to use.  As of right now, we cannot rely or "cheat" as we discussed by passing in an actionURL to `edit_commerce_order` because of this potential blocker.